### ISSUE TITLE
why it tell me the warning (SVD did not converge) when  i train the model used my data

### ISSUE THREAD

**Issue by YS-IMTech** [chenhsuanlin/bundle-adjusting-NeRF#64]
why it tell me the warning (SVD did not converge) when  i train the model used my data

**Below for details:**
![image](https://user-images.githubusercontent.com/65273749/208366669-cb92c984-4604-4d24-b05e-b4a54ad6936f.png)

**Comment by chenhsuanlin** [comment]
It is because Procrustes analysis failed (see [this line](https://github.com/chenhsuanlin/bundle-adjusting-NeRF/blob/803291bd0ee91c7c13fb5cc42195383c5ade7d15/model/barf.py#L113)). It could be because there is only one point being aligned, all the points are completely overlapping (resulting in a division-by-zero), or other causes. Hope this helps!

**Comment by jonathanhyunmoon** [comment]
Is this warning a problem for training? It persists throughout the training. 
Here are my loss and psnr curves.

![Screenshot from 2023-01-18 09-25-28](https://user-images.githubusercontent.com/12516283/213196734-ec6953e4-65d8-4fb3-8ee6-e13e2d503048.png)

![Screenshot from 2023-01-18 09-24-31](https://user-images.githubusercontent.com/12516283/213196540-78d66269-c695-462d-96cc-e9168f66f89b.png)

**Comment by chenhsuanlin** [comment]
Hi @jonathanhyunmoon, if you are running train-from-scratch experiments, it will raise the warning at the very beginning as the initial poses are completely overlapping (all identity), thus resulting in a division-by-zero. It shouldn't happen again once the training starts. If that's the case for you, then it should be fine. For @YS-IMTech's case, there seem to be other issues.

**Comment by jonathanhyunmoon** [comment]
Hi @chenhsuanlin, thanks for the quick response.

You are correct, I am working with "train-from-scratch" experiments" (i.e. the problem of unposed images vs. imperfect posed images). However, throughout all 200,000 iterations, the warning "warning: SVD did not converge..." appears at every validation step. You're saying it should only happen at the beginning correct?

Perhaps there is something else wrong with my training setup on my custom [data](https://drive.google.com/file/d/1YMBwQbPUzGlVWEmdAyHXyHLkApBec32Q/view?usp=sharing)?

Here are my resulting rgb and depth videos. 

- [rgb video](https://youtu.be/xxPFKtJaxPM)

- [depth video](https://youtu.be/DAyHFRAi8CA)

I'm wondering if the poor results and terrible validation curves (my validation PSNR curve is trending down and my validation loss is trending up) is correlated with this warning.

I'd appreciate any advice :) Thank you

**Comment by chenhsuanlin** [comment]
Closing due to inactivity, please feel free to reopen if necessary.

**Comment by curryandklay** [comment]
I downloaded the nerf_real_360 zipfile from NeRF Google Drive. I put the folder into data and renamed it to 'iphone', intending to use the pinecone scene for training; the training command is: 
`python3 train.py --group=mytest - -model=barf --yaml=barf_iphone --name=real_pinecone --data.scene=pinecone --barf_c2f=[0.1,0.5] --gpu=1`

when trying for the first time, an error message shows that the file name format is unrecognizable symbol (I guess it's the "_" symbol), so I removed "IMG_" from all image names and kept only the numbers; the second try gave the same warning as @YS-IMTech:
![image](https://github.com/chenhsuanlin/bundle-adjusting-NeRF/assets/121752440/f400bba8-dd0e-4d07-a480-18e36ade2369)

How can I find the cause of the problem? I would appreciate your reply, it would be very helpful!
